airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -26,18 +34,100 @@ def delete_dag_run():
     raise NotImplementedError("Not implemented yet.")
 
 
-def get_dag_run():
+@provide_session
+def get_dag_run(dag_id, dag_run_id, session):
     """
     Get a DAG Run.
     """
-    raise NotImplementedError("Not implemented yet.")
+    query = session.query(DagRun)
+    query = query.filter(DagRun.dag_id == dag_id)
+    query = query.filter(DagRun.run_id == dag_run_id)
+    dag_run = query.one_or_none()
+    if dag_run is None:
+        raise NotFound("DAGRun not found")
+    return dagrun_schema.dump(dag_run)
 
 
-def get_dag_runs():
+@provide_session
+def get_dag_runs(dag_id, session):
     """
     Get all DAG Runs.
     """
-    raise NotImplementedError("Not implemented yet.")
+    offset = request.args.get(parameters.page_offset, 0)
+    limit = min(int(request.args.get(parameters.page_limit, 100)), 100)
+
+    start_date_gte = parse_datetime_in_query(
+        request.args.get(parameters.filter_start_date_gte, None)
+    )
+    start_date_lte = parse_datetime_in_query(
+        request.args.get(parameters.filter_start_date_lte, None)
+    )
+
+    execution_date_gte = parse_datetime_in_query(
+        request.args.get(parameters.filter_execution_date_gte, None)
+    )
+    execution_date_lte = parse_datetime_in_query(
+        request.args.get(parameters.filter_execution_date_lte, None)
+    )
+
+    end_date_gte = parse_datetime_in_query(
+        request.args.get(parameters.filter_end_date_gte, None)
+    )
+    end_date_lte = parse_datetime_in_query(
+        request.args.get(parameters.filter_end_date_lte, None)
+    )
+
+    query = session.query(DagRun)
+
+    #  This endpoint allows specifying ~ as the dag_id to retrieve DAG Runs for all DAGs.
+    if dag_id == '~':
+        dag_run = query.all()
+        return dagrun_collection_schema.dump(DAGRunCollection(
+            dag_runs=dag_run,
+            total_entries=len(dag_run))
+        )
+
+    query = query.filter(DagRun.dag_id == dag_id)
```

Comment:
```suggestion
    if dag_id != '~':
        query = query.filter(DagRun.dag_id == dag_id)
```
We still want to have filters and paginations. ~ is a wildcard. It allows any value, so we should give up filtering in this field.